### PR TITLE
gemified + remastered some version history

### PR DESCRIPTION
**Pull request by kares**
Here's a second attempt to making _redjs_ something of a "releasable" gem.

Let me know if it does suit your taste or propose any changes so I can rebase it as needed.

The history kinda mimics the difficulities I did recall with `rhino` (I also noticed there where more specs failing than it seemed due a rhino bug) - `therubyrhino` should be fine with **0.2.1** but I shall manage **0.4.1** ... `v8` should be just fine.

If it's fine I'll setup tags - we can than immediately use it from bundler even without releasing the gem.
There's 5 specs still missing and 1ebcd2a5766cec58f93d80308682ff7858320d3e I'd like to add those as separate pulls on top when this gets sorted out - so we can have a discussion as I have some minor questions.

Also some of the #3 specs got incorporated.

### CODE REVIEW

File: lib/redjs/spec/context_spec.rb [hunk 1]
```diff
@@ -714,35 +728,39 @@ def cls.foo
...
+      @cxt.eval('obj2 instanceof Class1').should == true
+    end
+    
     it "unwraps instances created by a native constructor when passing them back to ruby" do
-      cls = Class.new.tap do |c|
-        c.class_eval do
-          def definitely_a_product_of_this_one_off_class?
-            true
-          end
+      @cxt['RubyClass'] = Class.new do
```
Comment (cowboyd) [inline lib/redjs/spec/context_spec.rb:756]: does passing a block to Class.new work in all versions of Ruby?  I never knew that.

File: lib/redjs/spec/context_spec.rb [hunk 2]
```diff
@@ -714,35 +728,39 @@ def cls.foo
...
+        def definitely_a_product_of_this_one_off_class?
+          true
         end
       end
-      @cxt['RubyClass'] = cls
       @cxt.eval('new RubyClass()').should be_definitely_a_product_of_this_one_off_class
     end
 
-    it "does not allow you to call a native ruby constructor, unless that constructor has been directly embedded" do
```
Comment (cowboyd) [inline lib/redjs/spec/context_spec.rb:739]: Any reason why this is removed? It seems important for security.

**Comment by cowboyd** [comment]
Thanks man!

Looks good, although I've noticed that there are some specs which have been removed entirely? Perhaps we should move them somewhere else so that they can be placed back in for later versions, or start breaking out different functional areas. 

Also, I noticed that the you use Context.open everywhere instead of Context.new V8::Context does not actually have an open method.

File: lib/redjs/spec/context_spec.rb (hunk 2 above)
Comment (kares) [inline lib/redjs/spec/context_spec.rb:739]: those are the specs I mentioned I'll add as separate pulls ... as soon as this gets sorted out. 
I only removed them due rhino as that ain't implemented ...

**Comment by kares** [comment]
Oh, I'm sorry about `open` that's yet another rhino-ism ... I just noticed as doing this refactoring that some specs where only passing as they did not execute the block passed to `new` and since I thought there's an `open` in TRR as well, since I do not remember adding it. I'll revert the commits to not contain any Context.open and update rhino instead ...

File: lib/redjs/spec/context_spec.rb (hunk 2 above)
Comment (cowboyd) [inline lib/redjs/spec/context_spec.rb:739]: Cool. As long as we have them stashed somewhere to be re-added later.

**Comment by kares** [comment]
commits are rewriten - no `open` anywhere - last call before getting this on master ...
don't worry about the missing specs I make sure nothing gets lost along the way :)

**Pull request merged**